cohortextractor2/main.py

```diff
@@ -215,7 +219,20 @@ def added_to_path(directory):
         sys.path = original
 
 
-def extract(cohort_class, backend):
+def extract(
+    cohort_class: Union[Cohort, type], backend: BaseBackend
+) -> Generator[dict[str, str], None, None]:
+    """
+    Extracts the cohort from the backend specified
+
+    Args:
+        cohort_class: The definition of the Cohort
```

Comment:
Since you've discovered that this name doesn't match the actual usage (it's _either_ a class, for the old DSL, or a `Cohort` instance, for the new DSL), I suggest that you rename this variable to reflect that (and the annotated type). I think `cohort_definition` would be better.

Reply:
I have named this `cohort_definition`